### PR TITLE
Send statistics to Geckoboard

### PR DESCRIPTION
**Pull request by pezholio**
This PR sets up the necessary plumbing for us to record statistics in Geckoboard, using the [Geckoboard Datasets API](https://www.geckoboard.com/datasets/). We have two datasets:

* `claims.submitted.$ENV` - This gives us a record of all the claims that have been submitted, which will allow us to get a count of claims submitted during a particular time period, as well as track activity over time. This dataset is updated every time a user submits a claim.
* `claims.paid.$ENV` - This gives us a record of all the claims that have been paid, which will allow us to get a count of claims that have been paid during a particular time period, as well as track activity over time. This is updated every time a claim is paid.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog]
 
 - Introduce the beginnings of the Maths & Physics journey, protected with basic
   auth
+- Send statistics on claims that have been submitted and paid to Geckoboard
```
Comment (tekin) [inline CHANGELOG.md]:
```suggestion
- Send statistics on submitted claims to Geckoboard
```

File: app/jobs/record_submitted_claim_job.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "geckoboard"
+
+class RecordSubmittedClaimJob < ApplicationJob
+  def perform(claim)
+    Claim::GeckoboardEvent.new(claim, :submitted, claim.submitted_at).record
```
Comment (tekin) [inline app/jobs/record_submitted_claim_job.rb]: This reads much nice than `RecordClaimEvent.new(...).perform` we had before 👍

File: azure/resource_groups/app/parameters/development.template.json [hunk 3]
```diff
@@ -156,6 +156,14 @@
         "secretName": "DfeSignInApiEndpoint"
       }
     },
+    "GECKOBOARD_API_KEY": {
+      "reference": {
+        "keyVault": {
+          "id": "${keyVaultId}"
+        },
+        "secretName": "GeckoboardAPIKey"
```
Comment (tekin) [inline azure/resource_groups/app/parameters/development.template.json]: Let's make sure this is added to both Azure environments before we merge.

File: azure/resource_groups/app/template.json [hunk 4]
```diff
@@ -153,7 +156,7 @@
     "ROLLBAR_ACCESS_TOKEN": {
       "type": "securestring"
     },
-    "ROLLBAR_ENV": {
+    "ENVIRONMENT_NAME": {
```
Comment (tekin) [inline azure/resource_groups/app/template.json]: 👍

File: app/jobs/record_submitted_claim_job.rb (hunk 2 above)
Comment (pezholio) [reply app/jobs/record_submitted_claim_job.rb]: 👍

**Pull request merged**